packages/frontend/src/pages/EventLogs/components/EventLogsActions.tsx

```diff
@@ -0,0 +1,198 @@
+import {
+  eventLogsQueryParameters,
+  useAllEventTypeLogs,
+} from '@/model/eventlogs';
+import { faFilter, faSearch } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Listbox, Transition } from '@headlessui/react';
+import { debounce } from '@mui/material';
+import { Fragment, useEffect, useMemo, useState } from 'react';
+import { CheckmarkIcon } from 'react-hot-toast';
+import { useRecoilState } from 'recoil';
+
+const sortOptions = [
+  { id: 'desc', label: 'Newest' },
+  { id: 'asc', label: 'Oldest' },
+];
+
+interface filterOptionsProps {
+  id: number;
+  label: string;
+}
+
+interface sortOptionsProps {
+  id: string;
+  label: string;
+}
+
+const EventLogsActions = (): JSX.Element => {
+  const [selectedFilters, setSelectedFilters] = useState<filterOptionsProps[]>(
+    []
+  );
+  const [selectedSort, setSelectedSort] = useState<sortOptionsProps>(
+    sortOptions[0]
+  );
+  const [searchValue, setSearchValue] = useState<string>('');
+
+  const { types } = useAllEventTypeLogs();
+  const filterOptions = types.docs;
+
+  const [queryParameters, setQueryParameters] = useRecoilState(
+    eventLogsQueryParameters
+  );
+
+  const returnValues = useMemo(
```

Comment:
getting a lint error

Reply:
> hm, i am not sure what to do here - if i add dependencies for useMemo hook (queryParameters & setQueryParameters) i will create infinite loop because we are updating queryParameters here and they will change every time. Also i am not really sure why component is rerendering every time I update queryParameters @mattyg

Sorry I think I didn't look closely enough at this the first time. I think the confusion comes from using this pattern where EventLogsList reads+writes to recoil state, and EventLogsActions also reads+writes to recoil state, and the parameters to the api request are stored in recoil state.

I think it would be clearer if instead:
- EventLogsList was the "smart" component, that reads+writes to recoil state, and stores the query parameters *internally* in it's component state
- while EventLogsActions was a "dumb" component, that purely takes in user input and emits it to it's parent component.  Or alternatively, could be simply merged into it's parent component.

Does that make sense? We could pair program it too if you want.